modules/@apostrophecms/notification/index.js

```diff
@@ -218,34 +229,36 @@ module.exports = {
         before: '@apostrophecms/global',
         middleware: async (req, res, next) => {
           let start;
-          let displayingIds;
           try {
-            if (req.method.toUpperCase() !== 'GET' || req.url !== self.action) {
+            const pathname = url.parse(req.url).pathname;
+            if (req.method.toUpperCase() !== 'GET' || pathname !== self.action) {
               return next();
             }
             if (!(req.user && req.user._id)) {
               throw self.apos.error('invalid');
             }
             start = Date.now();
-            displayingIds = self.apos.launder.ids(req.body.displayingIds);
             await attempt();
           } catch (e) {
-            return self.apiRouteSendError(res, e);
+            return self.apos.error(res, e);
```

Comment:
Since this is middleware returning won't respond and the request will hang. You need to actually send the error.

This will work:

`return self.routeSendError(req, e);`

Reply:
Done